tests/net/socket/tcp/src/main.c

```diff
@@ -368,6 +370,90 @@ static void restore_packet_loss_ratio(void)
 		"Error setting packet drop rate");
 }
 
+void test_v4_interface_down(void)
```

Comment:
The name is a bit misleading, the interface was not put down (i.e. `net_if_down()`), just the communication has been artificially blocked.

Reply:
What do you think about: test_v4_broken_link

Reply:
sounds better